src/ensembl/src/content/app/entity-viewer/gene-view/components/transcripts-filter/TranscriptsFilter.tsx

```diff
@@ -63,29 +70,35 @@ const radioData: RadioOptions = sortingOrderToLabel.map(([key, value]) => ({
   label: value
 }));
 
+export const metadataFields = ['biotype', 'appris', 'tsl'] as const;
+
+const createFilter = (metadataType: Filter['type'], label: string) => ({
+  label,
+  selected: false,
+  type: metadataType
+});
+
 const TranscriptsFilter = (props: Props) => {
   const filters = useSelector(getFilters);
   const sortingRule = useSelector(getSortingRule);
   const isSidebarOpen = useSelector(isEntityViewerSidebarOpen);
   const dispatch = useDispatch();
 
-  const biotypes = props.transcripts
-    .map((a) => a.so_term)
-    .filter(Boolean) as string[];
-
-  const uniqueBiotypes = Array.from(new Set(biotypes)).sort((a, b) => {
-    return sortBiotypes(a, b);
-  });
+  const initialFilters: Filters = {};
+
+  for (const transcript of props.transcripts) {
+    metadataFields.forEach((key) => {
+      const metadataItem = transcript.metadata[key];
+      if (metadataItem) {
+        initialFilters[metadataItem.value as string] = createFilter(
+          key,
+          metadataItem.label
+        );
+      }
+    });
+  }
```

Comment:
Maybe move this out of the component as well? So that the code in the component would just say `const initialFilters = createInitialFilters(props.transcripts)`?